src/middlewared/middlewared/plugins/enterprise/m_series_nvdimm_freebsd.py

```diff
@@ -0,0 +1,80 @@
+# -*- coding=utf-8 -*-
+import itertools
+import json
+import logging
+import os
+import re
+import struct
+
+import sysctl
+
+from middlewared.service import CallError, private, Service
+
+logger = logging.getLogger(__name__)
+
+
+class EnterpriseService(Service):
+
+    DATA = None
+    ERROR = "Data not retrieved yet"
+
+    @private
+    def setup_m_series_nvdimm(self):
+        try:
+            # This should be ran before pool import so we cache the value obtained on first middlewared startup
+            cache_path = "/tmp/m-series-nvdimm"
+            if os.path.exists(cache_path):
+                with open(cache_path) as f:
+                    self.DATA = json.load(f)
+                    return
+
+            result = []
+
+            for i in itertools.count(0):
+                try:
+                    pnpinfo = sysctl.filter(f"dev.nvdimm.{i}.%pnpinfo")[0].value
+                except IndexError:
+                    break
+
+                if "vendor=0x2c80 device=0x4e32" not in pnpinfo:
```

Comment:
@amotin please invite here someone who can check this

Reply:
I was right, after discussing with Nick Wolff it should be: "vendor=0x2c80 device=0x4e36" for 16GB and "vendor=0x2c80 device=0x4e33" for 32GB.  I don't think you need to check capacity after that.